### PR TITLE
[engine] Add ability to load aliases from resources

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: docs/modules/ROOT/pages/tests-configuration.adoc [hunk 1]
```diff
@@ -649,3 +649,51 @@ NOTE: `MYPAS_WORD` is a sample password and should be never used, own strong pas
...
 http.auth.password=ENC(WVnXhzG704KRZZFYL7hfpA4YoSm8V2F5eNXyr7CRYCichteHvNS3nB23zaI8CuLy)
 ----
+
+== User-defined aliases
+
+VIVIDUS allows to define set of aliases for steps in JSON format and then use them in user stories.
+
+=== How to use user-defined aliases in VIVIDUS
+
+. Select a step you want to create alias for, as an example let's take the following step:
```
Comment (valfirst) [inline docs/modules/ROOT/pages/tests-configuration.adoc:659]:
```suggestion
== Steps aliases

VIVIDUS allows to define set of aliases for steps in JSON format and then use them in stories.

=== How to use steps aliases

. Select a step you want to create alias for, as an example let's take the following step:
```

File: docs/modules/ROOT/pages/tests-configuration.adoc [hunk 2]
```diff
@@ -649,3 +649,51 @@ NOTE: `MYPAS_WORD` is a sample password and should be never used, own strong pas
...
++
+.VIVIDUS step
+[source,gherkin]
+----
+When I hover mouse over element located `$locator`
+----
+
+. Create JSON file at `aliases/aliases.json` (example) path and add the following JSON containing one or more aliases:
```
Comment (valfirst) [inline docs/modules/ROOT/pages/tests-configuration.adoc:667]:
```suggestion
. Create a new JSON file (for example, `aliases/aliases.json`) and put the following content with one or more aliases:
```

File: docs/modules/ROOT/pages/tests-configuration.adoc [hunk 3]
```diff
@@ -649,3 +649,51 @@ NOTE: `MYPAS_WORD` is a sample password and should be never used, own strong pas
...
++
+.aliases/aliases.json
+[source,json]
+----
+[
+    {
+        "name": "When I hover mouse over element located `$locator`",
+        "aliases": [
```
Comment (valfirst) [inline docs/modules/ROOT/pages/tests-configuration.adoc:675]: use a single brackets notation within the example

File: docs/modules/ROOT/pages/tests-configuration.adoc [hunk 4]
```diff
@@ -649,3 +649,51 @@ NOTE: `MYPAS_WORD` is a sample password and should be never used, own strong pas
...
+        ]
+    }
+]
+----
+
+. Point the alias configuration property to an exact location or location pattern with the JSON files containing ailases:
++
+[source,properties]
+----
+bdd.configuration.alias-paths=aliases/aliases.json
```
Comment (valfirst) [inline docs/modules/ROOT/pages/tests-configuration.adoc:691]:
```suggestion
engine.alias-paths=aliases/aliases.json
```

File: vividus-engine/src/main/resources/org/vividus/engine/spring.xml [hunk 5]
```diff
@@ -56,6 +56,7 @@
         <property name="storyReporterBuilder" ref="storyReporterBuilder" />
         <property name="placeholderResolver" ref="placeholderResolver" />
         <property name="compositePaths" value="${bdd.configuration.all-composite-paths}"/>
+        <property name="aliasPaths" value="${bdd.configuration.alias-paths}"/>
```
Comment (valfirst) [inline vividus-engine/src/main/resources/org/vividus/engine/spring.xml:59]:
```suggestion
        <property name="aliasPaths" value="${engine.alias-paths}"/>
```

File: vividus-tests/src/main/resources/story/integration/Aliases.story [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+Meta:
+    @epic vividus-engine
+    @feature aliases
+
+Scenario: Verify aliases loaded from resources
+
+!-- Vividus step
+Then `1` matches `\d+`
+!-- User-defined alias
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/Aliases.story:11]:
```suggestion
!-- Step alias configured in JSON
```

File: vividus-tests/src/main/resources/story/integration/Aliases.story [hunk 7]
```diff
@@ -0,0 +1,12 @@
+Description: Integration tests for aliases.
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/Aliases.story:1]:
```suggestion
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-tests/src/main/resources/story/integration/Aliases.story [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+Meta:
+    @epic vividus-engine
+    @feature aliases
+
+Scenario: Verify aliases loaded from resources
+
+!-- Vividus step
+Then `1` matches `\d+`
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/integration/Aliases.story:10]: is it possible to use a different language as an alias? f

Comment (uarlouski) [reply vividus-tests/src/main/resources/story/integration/Aliases.story:8]: some adjustment will be required
https://jbehave.org/reference/latest/stories-in-your-language.html

**Review by ikalinin1 — APPROVED** [review]

File: vividus-tests/src/main/resources/story/integration/Aliases.story (hunk 8 above)
Comment (ikalinin1) [reply vividus-tests/src/main/resources/story/integration/Aliases.story:8]: Does it make sense to create an issue for this? @valfirst @uarlouski

Comment (uarlouski) [reply vividus-tests/src/main/resources/story/integration/Aliases.story:8]: https://github.com/vividus-framework/vividus/issues/2498

**Pull request merged**